### DISCUSSION TITLE
Hono middlewares

### DISCUSSION THREAD

**Discussion by kyrylobashtenko** [web-infra-dev/modern.js#6607] [General]
Hi, can I use hono middlewares in modern.js ?
For example this one https://hono.dev/docs/middleware/builtin/cors, so i won't need any custom soltuions?
Thanks

**Comment by zllkjc** [comment]
can this features help ? https://modernjs.dev/apis/app/runtime/web-server/unstable_middleware.html

**Reply by kyrylobashtenko** [reply]
@zllkjc My problem is that I currently have an Express app, and we have already passed multiple penetration tests.
So ideally I wanted to achieve similar behavior for both CORS and CSP.
```
import cors from "cors";
import helmet from "helmet";
// CORS
app.use(cors(corsConfig));
// CSP
app.use(helmet(cspConfig));
```

But unfrotunalty UnstalbeMiddleware doesn't handle all static resources, maybe what I want is overkill, but I prefer to have the option to do that in case something is requested as a result of penetration testing.
That's why I was asking about[ Custom Web Server ](https://github.com/web-infra-dev/modern.js/discussions/4226)as well :)

**Comment by zllkjc** [comment]
We already support Hono Middleware.

You can refer to [New Custom Server](https://modernjs.dev/guides/advanced-features/custom-server.html).
